frontends/web/src/api/devices.ts

```diff
@@ -0,0 +1,282 @@
+
+type TProuctName = 'bitbox' | 'bitbox02' | 'bitbox02-bootloader';
+
+export type TStatus = 'connected'
+    | 'unpaired' | 'pairingFailed'
+    | 'uninitialized' | 'seeded' | 'initialized'
+    | 'require_firmware_upgrade' | 'require_app_upgrade';
+
+export type TStatusBB01 = 'unpaired' | 'pairingFailed'
+    | 'uninitialized' | 'seeded' | 'initialized'
+    | 'require_firmware_upgrade' | 'require_app_upgrade';
+
+type TSuccess = {
+    success: true;
+}
+
+// if the backend uses maybeBB02Err
+type TFail = {
+    code?: number;
+    message?: string;
+    success: false;
+}
+
+export type TDevices = {
+    readonly [key in string]: TProuctName;
+};
+
+export const getDeviceList = (): Promise<TDevices> => {
+    return apiGet('devices/registered');
+};
+
+export type TDeviceInfo = {
+    initialized: boolean;
+    mnemonicPassphraseEnabled: boolean;
+    name: string;
+    version: string;
+}
+
+/**
+ * getDeviceInfo returns device-info or a fail response,
+ * but does not provide a success info. For this reason
+ * the success property has to be used.
+ * For an example of a reponse that returns both success
+ * or fail, see setDeviceName.
+ *
+    getDeviceInfo('deviceId').then(deviceInfo => {
+        if ('success' in deviceInfo) {
+            // fail as deviceInfo.success = false
+        } else {
+            // success but deviceInfo.success is missing!
+        }
+    })
+ */
+
+export const getDeviceInfo = (
+    deviceID: string
+): Promise<TDeviceInfo | TFail> => {
+    return apiGet(`devices/bitbox02/${deviceID}/info`);
+};
+
+export type TDeviceInfoBB01 = {
+    bootlock: boolean;
+    id: string;
+    lock: boolean;
+    name: string;
+    new_hidden_wallet: boolean; // eslint-disable-line camelcase
+    pairing: boolean;
+    sdcard: boolean;
+    seeded: boolean;
+    serial: string;
+    TFA: string;
+    U2F: boolean;
+    U2F_hijack: boolean; // eslint-disable-line camelcase
+    version: string;
+}
+
+export const getDeviceInfoBB01 = (
+    deviceID: string,
+): Promise<TDeviceInfoBB01> => {
+    return apiGet(`devices/${deviceID}/info`);
+};
+
+/**
+ * Set device name of deviceID, returns success or fail.
+ *
+    setDeviceName('deviceid', 'myBB02').then(data => {
+        if (data.success) {
+            // success!
+        } else {
+            // fail
+        }
+    })
+ */
+
+export const setDeviceName = (
+    deviceID: string,
+    name: string,
+): Promise<TSuccess | TFail> => {
+    return apiPost(`devices/${deviceID}/set-device-name`, { name });
+};
+
+export const getStatus = (deviceID: string): Promise<TStatus> => {
+    return apiGet(`devices/${deviceID}/status`);
+};
+
+export const getStatusBB01 = (deviceID: string): Promise<TStatusBB01> => {
+    return apiGet(`devices/${deviceID}/status`);
+};
+
+// firmware
+
+export type TBundledFirmwareVersion = {
+    canUpgrade: boolean;
+    currentVersion: string;
+    newVersion: string;
+}
+
+export const getBundledFirmwareVersion = (
```

Comment:
many of these funcs in here seem unused. please remove them until they are needed, otherwise it's difficult to see if they are correct without being able to test their use.

Reply:
This time, I didn't copy from the existing code, but typed the api by reading go backend, so I need to back it up :)

FYI on this branch there is still unused `api/account.ts`, we should merge master into this branch.